wgpu-core/src/device/resource.rs

```diff
@@ -89,7 +88,7 @@ use super::{
 pub struct Device<A: HalApi> {
     raw: Option<A::Device>,
     pub(crate) adapter: Arc<Adapter<A>>,
-    pub(crate) queue_id: RwLock<Option<QueueId>>,
+    pub(crate) queue: RwLock<Option<Weak<Queue<A>>>>,
```

Comment:
This isn't a problem in this PR, but I think this is a problem in this design. I think we can get in a situation where the device exists but the queue has been dropped. I do think we need to re-think about how the device/queue is modeld.